lib/u2f.js

```diff
@@ -0,0 +1,186 @@
+/*!
+ * u2f.js - Ledger U2F communication
+ * Copyright (c) 2017, The Bcoin Developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+/* eslint-env browser */
+'use strict';
+
+const assert = require('assert');
+
+const U2F = require('u2f-api');
+const {sign} = U2F;
+const {Lock} = require('bmutex');
+
+const LedgerError = require('./error');
+const {Device, DeviceInfo} = require('./device');
+
+const U2F_VERSION = 'U2F_V2';
+const U2F_CHALLENGE = base64ToSafe(Buffer.alloc(32).toString('base64'));
+
+/**
+ * U2F Device
+ * @extends {Device}
+ */
+class U2FDevice extends Device {
+  /**
+   * Create Ledger U2F device
+   * @constructor
+   * @param {Object} options
+   */
+
+  constructor(options) {
+    super();
+
+    this.lock = new Lock(false);
+
+    if (options)
+      this.set(options);
+  }
+
+  /**
+   * Set device options
+   * @param {Object} options
+   */
+
+  set(options) {
+    assert(options);
+    super.set(options);
+
+    return this;
+  }
+
+  /**
+   * Checks if the browser supports U2F
+   * @throws {LedgerError}
+   */
+
+  async open() {
+    await U2FDevice.enforceSupport();
+  }
+
+  /**
+   * Mock function
+   */
+
+  async close() {
```

Comment:
why mock?

Reply:
u2f does not have close/open or connection. But if you have written code for `node-hid` you can just copy paste with this.

Reply:
So this is in case you're using `node-hid` code that calls a `close` method and we don't want it to throw an error that the method doesn't exist?

Reply:
yep, exactly. Basically you should be able to change Device (Driver) without code change.